dds/DCPS/SendStateDataSampleList.inl

```diff
@@ -130,10 +131,12 @@ ACE_INLINE
 void
 SendStateDataSampleList::remove(DataSampleElement* stale)
```

Comment:
Should this function decrement `size_`?

Reply:
This method is static and its description in the header file says it removes `stale` from whatever list object it belongs to. But you're right that `size_` of the affected list object must be decreased somewhere. I'm not sure if the callers of this method do the decrement. From a quick search, it seems to me that `size_` is not updated where this method is called. So this may be something to look further into.

Reply:
It looks like this method is only ever called by the WriteDataContainer when it can't figure out which list a "delivered" or "dropped" DataSampleElement came from... which I hope is a corner case we don't actually hit very often. Note in the comment description it admits that this invalidates the list this element comes from. As far as I'm concerned, this method shouldn't exist.

@mitza-oci Can you think of any reason why we couldn't remove this method (and it's calls) from the WriteDataContainer? Surely if we can't find the list a delivered/dropped sample came from, we should just log & ignore it rather than try to remove it from some list we don't own and/or that doesn't contain this sample anymore.

Reply:
The comments are referring to `get_resend_data()` which to me looks different than "can't find the list".  But I agree with the general point, if we want a different way to handle this it is only used in two places in WriteDataContainer.

Reply:
I think it's OK to leave it for now, but only because I'm guessing our future transport framework changes are going to have us revisiting the WriteDataContainer & associated lists as well.